### PR TITLE
Evaluate with macro f1

### PR DESCRIPTION
**Pull request by JoeCx**
# Overview

**Type of Change:** New Feature

**Summary:** Added Macro F1 Calculation for training_program.py and alt_training_program.py. Now when training is run, rather than saving a model based on its weighted f1 score, the model is saved based on its macro f1 score. This is better because now classes have equal representation in the accuracy evaluation.

## UI/UX Implementation

No changes made.

## Model Updates

This section describes changes that were made to the models used by your application.

### Data Models

No changes made.

### Object-Oriented Models

No changes made.

## End-to-End Testing Instructions

Run both training programs to see the new macro f1 scores for the model. Make sure to delete previously saved weighted f1 scores in the accuracy jsons for the models(spec_accuracies.json, gen_accuracies.json, alt_spec_accuracies.json, and alt_gen_accuracies.json).

### CODE REVIEW

File: src/alt_training_program.py [hunk 1]
```diff
@@ -173,11 +176,19 @@ def training_evaluation_dorsal_caudal(self, num_epochs, train_loader, test_loade
...
             accuracy = correct / total
-            self.model_accuracies["dors_caud"] = accuracy
             print(f"Accuracy: {100 * accuracy:.2f}%")
 
+            # Compute and print F1 scores
+            weighted_f1 = f1_score(all_labels, all_predictions, average='weighted')
+            macro_f1 = f1_score(all_labels, all_predictions, average='macro')
+            self.model_accuracies["dors_caud"] = macro_f1
+            print(f"Weighted F1 Score: {100 * weighted_f1:.2f}%")
+            print(f"Macro F1 Score: {100 * macro_f1:.2f}%")
```
Comment (tmckeown311) [inline src/alt_training_program.py:190]: Good idea using both to display the difference between the two scores. This should be helpful as data augmentation helps us moving forward

**Review by tmckeown311 — APPROVED** [review]
Good addition. The added metrics should increase our accuracy over time as we will be able to see two different approaches to evaluating our program and will be able to improve different aspects of the program this way
